packages/mui-material/src/PaginationItem/PaginationItem.js

```diff
@@ -334,19 +336,86 @@ const PaginationItem = React.forwardRef(function PaginationItem(inProps, ref) {
 
   const normalizedIcons = isRtl
     ? {
-        previous: slots.next || components.next || NavigateNextIcon,
-        next: slots.previous || components.previous || NavigateBeforeIcon,
-        last: slots.first || components.first || FirstPageIcon,
-        first: slots.last || components.last || LastPageIcon,
+        previous: slots.next ?? components.next,
+        next: slots.previous ?? components.previous,
+        last: slots.first ?? components.first,
+        first: slots.last ?? components.last,
       }
     : {
-        previous: slots.previous || components.previous || NavigateBeforeIcon,
-        next: slots.next || components.next || NavigateNextIcon,
-        first: slots.first || components.first || FirstPageIcon,
-        last: slots.last || components.last || LastPageIcon,
+        previous: slots.previous ?? components.previous,
+        next: slots.next ?? components.next,
+        first: slots.first ?? components.first,
+        last: slots.last ?? components.last,
       };
 
-  const Icon = normalizedIcons[type];
+  const externalForwardedProps = {
+    slots: { ...normalizedIcons, ...slots },
```

Comment:
The RTL implementation is getting hard to follow. What do you think about refactoring so the shuffle happens at the end:

```jsx
const externalForwardedProps = {
    slots: {
        previous: slots.previous ?? components.previous,
        next: slots.next ?? components.next,
        first: slots.first ?? components.first,
        last: slots.last ?? components.last,
    },
    slotProps
}


const [PreviousSlot, previousSlotProps] = useSlot('previous', {
    elementType: NavigateBeforeIcon,
    externalForwardedProps,
    ownerState,
});

const [NextSlot, nextSlotProps] = useSlot('next', {
    elementType: NavigateNextIcon,
    externalForwardedProps,
    ownerState,
});

const [FirstSlot, firstSlotProps] = useSlot('first', {
    elementType: FirstPageIcon
    externalForwardedProps,
    ownerState,
});

const [LastSlot, lastSlotProps] = useSlot('last', {
    elementType: LastPageIcon,
    externalForwardedProps,
    ownerState,
});

const rtlAwareType = isRtl ? {
    previous: 'next',
    next: 'previous',
    first: 'last',
    last: 'first',
}[type] : type;

const IconSlot = {
    previous: PreviousSlot,
    next: NextSlot,
    first: FirstSlot,
    last: LastSlot,
}[rtlAwareType];

const iconSlotProps = {
    previous: previousSlotProps,
    next: nextSlotProps,
    first: firstSlotProps,
    last: lastSlotProps,
}[rtlAwareType];
```

This way, the shuffle happens in only one place, and we have a specific variable named for it.

Do we have tests for this RTL shuffle?

Reply:
>Do we have tests for this RTL shuffle?

added here https://github.com/mui/material-ui/pull/41777/commits/606f8c323857e1d40e6e03045e74e7ba1c9bffe2

refactor commit -> https://github.com/mui/material-ui/pull/41777/commits/f26c8529929e115959ed9dbafb879722290ba856